src/plugins/image-extension/edit.js

```diff
@@ -0,0 +1,189 @@
+/** @jsx jsx */
+/**
+ * External dependencies.
+ */
+import hexToRgba from 'hex-rgba';
+
+window.hexToRgba = hexToRgba;
+
+import {
+	css,
+	jsx
+} from '@emotion/react';
+
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+
+import {
+	ColorPalette,
+	InspectorControls
+} from '@wordpress/block-editor';
+
+import {
+	PanelBody,
+	RangeControl,
+	ToggleControl
+} from '@wordpress/components';
+
+import { useSelect } from '@wordpress/data';
+
+import {
+	Fragment,
+	useEffect
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import ColorBaseControl from '../../components/color-base-control/index.js';
+import ControlPanelControl from '../../components/control-panel-control/index.js';
+
+const Edit = ({
+	BlockEdit,
+	props
+}) => {
+	const { attributes, setAttributes } = props;
+
+	const IDs = useSelect( select => select( 'core/block-editor' ).getBlocks().filter( block => 'core/image' === block.name && undefined !== block.attributes.anchor ).map( block => block.attributes.anchor ) );
+
+	useEffect( () => {
+		if ( attributes.boxShadow ) {
+			const isUnique = attributes.anchor ? ( 1 === IDs.filter( id => id === attributes.anchor ).length ) : false;
+
+			if ( ! isUnique ) {
+				const anchor = `wp-block-themeisle-blocks-image-${ uuidv4().substr( 0, 8 ) }`;
+				setAttributes({ anchor });
+			}
+		} else {
+			if ( attributes.anchor && attributes.anchor.includes( 'wp-block-themeisle-blocks-image-' ) ) {
+				const anchor = undefined;
+				setAttributes({ anchor });
+			}
+		}
+	}, [ attributes.boxShadow ]);
+
+	const changeBoxShadowColor = value => {
+		setAttributes({ boxShadowColor: ( 100 > attributes.boxShadowColorOpacity ) ? getComputedStyle( document.documentElement, null ).getPropertyValue( value?.replace( 'var(', '' )?.replace( ')', '' ) ) : value });
+	};
+
+	const changeBoxShadow = value => {
+		setAttributes({ boxShadow: value });
+	};
+
+	const changeBoxShadowColorOpacity = value => {
+		const changes = { boxShadowColorOpacity: value };
+		if ( 100 > value && attributes.boxShadowColor?.includes( 'var(' ) ) {
```

Comment:
@Soare-Robert-Daniel seems like you convert a css var into a hardcoded value. What happens when the CSS var is changed? WIll the block use the new color or the previous value of the css var?

Reply:
It will be the previous value. With javascript, this is not much of a problem because we can access the CSS var value anytime via evaluation. The problem is with PHP; we do not have this luxury. This is the trade-off of this technic.

Reply:
why we can't use the css var as a shadow color instead of the hex value?

Reply:
No idea. The previous implementation used this system with transformation and did not create big concerns, as I am concern. So I kept the same idea.

But looking at this https://stackoverflow.com/questions/12216572/add-alpha-channel-to-a-given-color
The whole thing with using the hex transformation might have been useless.

Reply:
But I am sure if @HardeepAsrani might have been considered this and run into an edge case when he first implemented the shadow picker for advanced heading.

Reply:
After some testing, I can not use something like this `rgba(var(--nv-c-1), 0.57)` where `--nv-c-1: #9463ae;`

The var needs to be like this: `--color_rgb: 0, 0, 0;` like a tuple of 3 values and not a single hex value.
This feature is present in SASS/SCSS but in CSS.

Reply:
Yea, rgba works function in CSS works when the variable is rgb value, not with hex. We can't add opacity to a CSS variable without converting it to a static value. @selul - when I first implemented the shadow option, we weren't so much testing with variables but normal colors. I don't remember but I guess Neve wasn't using CSS vars then? The issue never came up in any testing or users.

Reply:
ok, is there any other mechanism to add shadow outside of the box-shadow css prop? Otherwise, we can keep it like this.

Reply:
Nope, unless we use JS for this which so far we don't load.

Reply:
@selul @Soare-Robert-Daniel I found something which is similar to box-shadow but is made particularly for images: https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/drop-shadow()

It's like drop-shadow in Photoshop. It works well and can work with variables too. The only issue here is that if you use a square PNG image of a round object, the shadow will be round too, not square. It ignores the transparent part of the image.

What do you think?

Reply:
It has the same problem with `opacity,` but I do not know if it is better on the design part.

Reply:
we can review in neve a shift from hex to RGB to support this in the future, for now, no need to further bother with this.

Reply:
Sure, that can work. Then we can add a condition to not convert if the variable selected is currently in RGB.

Reply:
@selul From hex to RGB you mean something like this: `--nv: rgb(120, 120, 120)` then the problem will remain.

With that format, we can not plug & play something like this `rgba(var(--nv), opacity)`.

One idea is to use a SASS engine that can enable that feature. But is not too feasible.